langlib/lang.value/src/main/ballerina/value.bal

```diff
@@ -80,6 +94,20 @@ public isolated function cloneReadOnly(CloneableType  v) returns CloneableType &
 # - if a record type descriptor specifies default values, these will be used
 #   to supply any missing members
 #
+# ```ballerina
+# int[] arr = [1, 2, 3, 4];
+# int[] numberArray = check arr.cloneWithType();
```

Comment:
One would use `cloneWithType` instead of `clone` only if the type is different. So, `arr` should be something like an `anydata[]`.